Source/Optional.swift

```diff
@@ -1,20 +1,70 @@
+/**
+    map a function over an optional value
+
+    - If the value is .None, the function will not be evaluated and this will return .None
+    - If the value is .Some, the function will be applied to the unwrapped value
```

Comment:
Well, `map` applies a function to the value in a functor -- I think "applied" makes sense here.